lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -2574,6 +2575,15 @@ def main():
 
     result = {'failed': False, 'changed': False}
 
+    # Display deprecation warning for deprecated 'state' choices
+    if module.params['state'] in ('poweredoff', 'poweredon', 'rebootguest', 'shutdownguest'):
+        deprecated_state = module.params['state']
+        module.params['state'] = deprecated_state.replace('powered', 'powered-').replace('guest', '-guest')
+        module.deprecate(
+            msg="the use of '{0}' is deprecated, please switch to the new format, '{1}')".format(deprecated_state, module.params['state']),
+            version='2.13'
+        )
+
```

Comment:
Hi @Akasurde - Thanks for reviewing this PR. I was deprecating the old style and converting it to the new style. That was why I left out the old style (with the dash) from both the docs and the code.

Reply:
Yes, we will remove old style while remove it all together. Let me know if it works for you.

Reply:
Hi @Akasurde - based on that (deprecating the old style) do we need all of the other changes above? As the PR stands, docs will no longer mention the old style, and if you use the old style you will get a deprecation warning and it will be immediately converted to the new style (see the block this comment started on) so that we don't need to check for both styles later on.

Reply:
Hi @dericcrago, I'm not sure I fully understand your message. What do you mean by "other changes above".

Reply:
sorry @goneri & @Akasurde - that was confusing, I just fixed my comment, I meant "suggested changes above".